### DISCUSSION TITLE
Axis Cloning and Auto Squaring

### DISCUSSION THREAD

**Discussion by rvalotta** [grblHAL/core#18] [Q&A]
Looking for info on how to set this up. STM32F4 is the driver i am using and can't seem to find any info on it.   If it is not implemented on this driver, what driver does have it implemented.

**Comment by terjeio** [comment]
It is currently not implemented on this driver since it does not have any board map files that has the required two limit switch inputs per axis (to be auto squared) defined. Boards supporting > 3 axes may be reconfigured for this, but I have not spent time on that.

Currently some board maps for the iMXRT1062 and SAM3X8E has the option for cloning/autosquaring axes. It should not be too hard to add it for other drivers, but I do not own boards for all variants that would allow me to carry out tests for verification.

Which board are you using? See [this issue](https://github.com/grblHAL/STM32F4xx/issues/8) for a user planning to use the STM32F4 driver for a SKR PRO board.

**Reply by rvalotta** [reply]
I'm just using a black pill to drive my stepper drivers and limit switches.  As for the extra pin traditionally what i've seen done is that the Z switch is hooked up as the secondary axis limit switch. 

Can you provide me some direction on the axis cloning ? I will start to dig into the iMXRT1062 driver and see if i can figure any of this out.

**Reply by terjeio** [reply]
See [this issue](https://github.com/terjeio/grblHAL/issues/18) for some details.
